src/Data/Fin/Properties.agda

```diff
@@ -547,12 +547,21 @@ inject≤-idempotent {_} {suc n} {suc o} zero    _   _   _ = refl
 inject≤-idempotent {_} {suc n} {suc o} (suc i) (s≤s m≤n) (s≤s n≤o) (s≤s m≤o) =
   cong suc (inject≤-idempotent i m≤n n≤o m≤o)
 
+inject≤-trans : ∀ (i : Fin m) (m≤n : m ℕ.≤ n) (n≤o : n ℕ.≤ o) →
+                inject≤ (inject≤ i m≤n) n≤o ≡ inject≤ i (ℕₚ.≤-trans m≤n n≤o)
+inject≤-trans i m≤n n≤o = inject≤-idempotent i m≤n n≤o _
+
 inject≤-injective : ∀ (m≤n m≤n′ : m ℕ.≤ n) i j →
                     inject≤ i m≤n ≡ inject≤ j m≤n′ → i ≡ j
 inject≤-injective (s≤s p) (s≤s q) zero    zero    eq = refl
 inject≤-injective (s≤s p) (s≤s q) (suc i) (suc j) eq =
   cong suc (inject≤-injective p q i j (suc-injective eq))
 
+inject≤-irrelevant : ∀ (m≤n m≤n′ : m ℕ.≤ n) i →
+                    inject≤ i m≤n ≡ inject≤ i m≤n′
+inject≤-irrelevant (s≤s p) (s≤s q) zero    = refl
+inject≤-irrelevant (s≤s p) (s≤s q) (suc i) = cong suc (inject≤-irrelevant p q i)
```

Comment:
Why are we proving this by induction? Can't we do something like `cong (inject≤ i) (≤-irrelevant p1 p2)`

https://github.com/agda/agda-stdlib/blob/85829575904d9d7afc7a5ecc2b6902365f9cd564/src/Data/Nat/Properties.agda#L175

Reply:
Of course! Oh dear... I think I probably need to go back to school about the whole issue of `Irrelevant` propositions in the library, and the relation between *proving* irrelevance, and the typechecker being able to *observe* it.